Adding Everforest themes

mvykool opened the pull request:
Fixes #issue-number

## What does this PR do?

Adding Everforest themes, and updates an import from shiki, in file: https://github.com/webtui/webtui/blob/master/web/src/components/Example.astro

Replacing `getHighlighter` for `createHighlighter` as recommended in the documentation: https://shiki.matsu.io/blog/v2#gethighlighter-createhighlighter

## Checklist

- [x] I have read the [Contributing Guide](https://webtui.ironclad.sh/contributing/contributing)
- [x] My Pull Request abides by the [Style Guide](https://webtui.ironclad.sh/contributing/style-guide)

<img width="1684" alt="Screenshot 2025-05-09 at 5 52 09 AM" src="https://github.com/user-attachments/assets/e1327569-5822-437c-8d89-4973c1d8449d" />
<img width="1684" alt="Screenshot 2025-05-09 at 5 52 28 AM" src="https://github.com/user-attachments/assets/23de79cc-eebe-47a6-b6ec-9ece6a2265c7" />

IroncladDev reviewed (changes requested):
Added some comments

Make sure you use the token names from the actual everforest theme instead of reusing the gruvbox ones

There are some unnecessary diffs that can be fixed with `turbo run format`

Once you're finished addressing the changes, let me know and I'll do another pass

mvykool commented:
Thanks for the feedback, I addressed the comments you made @IroncladDev 
I did use the gruvbox theme as a baseline, since it has similar variants, but I overlooked some dumb things 

The last commit should fix those mistakes, let me know what you think

IroncladDev commented on packages/theme-everforest/src/base.css:
```diff
@@ -0,0 +1,153 @@
+  /* Base Dark Theme */
+  [data-webtui-theme|="everforest-dark"] {
+    /* Background colors */
+    --ef-dark-bg1: #2d353b;
```
Don't split out variables into dark/light

Instead of `--ef-dark-bg1` just do `--ef-bg1` or even just `--bg1` without the `ef-` prefix

IroncladDev commented on packages/theme-everforest/src/components/badge.css:
```diff
@@ -0,0 +1,34 @@
+  [data-webtui-theme|="everforest"] {
+    [is-~="badge"] {
+      &[variant-="red"] {
+        --badge-color: var(--gb-red);
```
Still using gruvbox vars

Change to `ef-` or remove the prefix from all vars alltogether

IroncladDev commented on packages/theme-everforest/src/components/button.css:
```diff
@@ -0,0 +1,48 @@
+  [data-webtui-theme|="everforest"] {
+    button {
+      &[variant-="red"] {
+        --button-primary: var(--gb-red);
```
Gruvbox vars

IroncladDev commented on packages/theme-everforest/src/components/typography.css:
```diff
@@ -0,0 +1,30 @@
+    h4,
+    h5,
+    h6 {
+      color: var(--gb-green);
```
Gruvbox vars

IroncladDev reviewed (changes requested):
Requested some more changes.

Please run `bun format` and test to ensure all themes work correctly before re-requesting review.

Looking good so far!

IroncladDev commented:
Bumping stale PR

Do you still want to work on this?

IroncladDev commented:
Alright, just checking. No need to hurry if you're busy.
